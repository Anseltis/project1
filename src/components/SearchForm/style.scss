%main-search-text-shared {
  font-family: Arial, sans-serif;
  text-transform: uppercase;
  color: #fff;
}

.main-search {
  padding: 0 6.25rem;

  &__h1 {
    @extend %main-search-text-shared;
    font-size: 2rem;
    font-weight: 600;
  }

  &__button {
    @extend %main-search-text-shared;
    font-size: 1.25rem;
    font-weight: 500;
    padding: .3125rem 2.1875rem;
    border: 0;
    background-color: #f00;


  }

  &__container {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
  }
}
