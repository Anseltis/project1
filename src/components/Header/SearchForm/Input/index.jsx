import React from 'react'
import './style.scss'

export const Input = ({className, placeholder, onChange, value}) => {
    return (
        <input className={className} type="text" placeholder={placeholder} onChange={onChange} value = {value}/>
    )
}
