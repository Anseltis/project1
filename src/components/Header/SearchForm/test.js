import React from 'react'

import { SearchForm } from 'components/Header/SearchForm/index'
import renderer from 'react-test-renderer'

test('renders Input properly', () => {
    const wrapper = renderer
        .create(<SearchForm/>)
        .toJSON()
    expect(wrapper).toMatchSnapshot()

})