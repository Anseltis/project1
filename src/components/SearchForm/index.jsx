import React from 'react'

import { Input } from './Input'
import { SearchByPanel } from '../SearchByPanel'

import './style.scss'

export const SearchForm = ({
  className,
  onChange,
  onSubmit,
  value,
  searchBy,
  searchByActions,
  filter
}) => {
  return (
    <form className={`${className} main-search`} onSubmit={onSubmit}>
      <h1 className="main-search__h1">Find your movie</h1>
      <Input
        className="main-search__input"
        placeholder="Quentin Tarantino"
        onChange={onChange}
        value={value}
      />
      <div className="main-search__container">
        <SearchByPanel
          className="search-by"
          searchBy={searchBy}
          searchByActions={searchByActions}
          filter={filter}
        />
        <input type="submit" className="main-search__button" />
      </div>
    </form>
  )
}
