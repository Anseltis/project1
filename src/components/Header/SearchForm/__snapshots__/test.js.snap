// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Input properly 1`] = `
<form
  className="undefined main-search"
>
  <h1
    className="main-search__h1"
  >
    Find your movie
  </h1>
  <input
    className="main-search__input"
    placeholder="Quentin Tarantino"
    type="text"
  />
  <div
    className="main-search__container"
  >
    <div
      className="search-by"
    >
      <span
        className="search-by__header"
      >
        Search by
      </span>
      <input
        className="search-by__title-input"
        id="title"
        name="search-by"
        type="radio"
        value="title"
      />
      <label
        className="search-by__title-label undefined"
        htmlFor="title"
      >
        Title
      </label>
      <input
        className="search-by__genre-input"
        id="genre"
        name="search-by"
        type="radio"
        value="genre"
      />
      <label
        className="search-by__genre-label undefined"
        htmlFor="genre"
      >
        Genre
      </label>
    </div>
    <input
      className="main-search__button"
      type="submit"
    />
  </div>
</form>
`;
